tracer/src/Datadog.Trace/DatabaseMonitoring/DatabaseMonitoringPropagator.cs

```diff
@@ -105,7 +107,87 @@ propagationLevel is not (DbmPropagationLevel.Service or DbmPropagationLevel.Full
+            {
+                /*
+                * For CommandType.StoredProcedure, we need to modify the command text to use an EXEC statement and swap it to a CommandType.Text.
+                * Without doing this we _cannot_ inject the comment as it will be taken as part of the stored procedure name.
+                *    in SqlCommand.cs we have: rpc.rpcName = this.CommandText; // just get the raw command text
+                * Injecting the comment in the CommandText would result in getting an exception that the StoredProcedure name wasn't found (and would look like an empty string).
+                *
+                * What this function does is:
+                *   - Swap to EXEC statement (which is a text command executing a stored procedure).
+                *   - Build a parameter list for the EXEC statement, so that we can pass all the parameters to it.
+                *   - These are the Input and Output parameters, but they may not be present
+                *   - We don't modify the parameters on the Command itself, this will still be handled fine by SQL Server from testing
+                * Some helpful links:
+                * reference RPC for stored procedure: https://github.com/microsoft/referencesource/blob/51cf7850defa8a17d815b4700b67116e3fa283c2/System.Data/System/Data/SqlClient/SqlCommand.cs#L5548
+                * reference text SQL (with params): https://github.com/microsoft/referencesource/blob/51cf7850defa8a17d815b4700b67116e3fa283c2/System.Data/System/Data/SqlClient/SqlCommand.cs#L4488
+                * reference text SQL (no params): https://github.com/microsoft/referencesource/blob/51cf7850defa8a17d815b4700b67116e3fa283c2/System.Data/System/Data/SqlClient/SqlCommand.cs#L4437
+                */
+                // Save the original stored procedure name
+                string procName = command.CommandText ?? string.Empty;
+
+                if (string.IsNullOrEmpty(procName))
+                {
+                    return false;
+                }
+
+                // Build parameter list for EXEC statement
+                var paramList = new StringBuilder();
+                foreach (DbParameter? param in command.Parameters)
+                {
+                    if (param == null)
+                    {
+                        continue;
+                    }
+
+                    // Skip return value parameters
+                    if (param.Direction == ParameterDirection.ReturnValue)
+                    {
+                        // we don't need to include this in the command text, it is handled in the SQLCommand
+                        continue;
+                    }
+
+                    if (paramList.Length > 0)
+                    {
+                        // e.g. @Input=@Input, @Input2=@Input2 OUTPUT
+                        paramList.Append(", ");
+                    }
+
+                    paramList.Append(param.ParameterName).Append('=').Append(param.ParameterName);
```

Comment:
I'm not really educated on SQL, but do we need it to be inserted as `@PARAM_NAME1=@PARAM_NAME1,@PARAM_NAME2=@PARAM_NAME2` or can it simply be inserted as `@PARAM_NAME1, @PARAM_NAME2` ?

Reply:
So you can do that, but it then requires them all to be in the same position as defined. This `=` way means that the order of the parameters isn't important

So I think that then risks something like the Parameter.Add order to be critical for users

`command.Parameter.Add(Param2)`
`command.Parameter.Add(Param1)`


I'm not 100% sure on this though, so I could test it out.

Reply:
That sounds safer, so if your proposed approach works then I don't see a need to test my suggestion